crates/sim2h/src/job/pool.rs

```diff
@@ -0,0 +1,55 @@
+use crate::*;
+
+/// a cpu-count pool of threads that can execute jobs
+pub(crate) struct Pool {
+    job_cont: Arc<Mutex<bool>>,
+    job_threads: Vec<std::thread::JoinHandle<()>>,
+    job_send: crossbeam_channel::Sender<Box<dyn Job>>,
+}
+
+impl Pool {
+    pub(crate) fn new() -> Self {
+        let (job_send, job_recv) = crossbeam_channel::unbounded::<Box<dyn Job>>();
+        let job_cont = Arc::new(Mutex::new(true));
+        let mut job_threads = Vec::new();
+        for _ in 0..num_cpus::get() {
+            let cont = job_cont.clone();
+            let send = job_send.clone();
+            let recv = job_recv.clone();
+            job_threads.push(std::thread::spawn(move || loop {
```

Comment:
Would be cool to record the cpu number in the thread id, not essential though.

Reply:
I like that idea, will do :+1: